autoupdate.js

```diff
@@ -0,0 +1,60 @@
+const autoUpdater = require('electron-updater').autoUpdater
+const { dialog } = require('electron');
+
+const hour = 60 * 60;
+const autoUpdaterInterval = hour * 1000;
+
+const RESTART_BUTTON = 0;
+const LATER_BUTTON = 1;
+
+function autoUpdateDisabled(config) {
+  return process.mas || config.get('disableAutoUpdate');
+}
+
+function checkForUpdates() {
+  autoUpdater.checkForUpdates();
+}
+
+function showUpdateDialog(localeMessages) {
+  return function() {
+    const options = {
+      type: 'info',
+      buttons: [
+        localeMessages.autoUpdateRestartButtonLabel.message,
+        localeMessages.autoUpdateLaterButtonLabel.message
+      ],
+      title: localeMessages.autoUpdateNewVersionTitle.message,
+      message: localeMessages.autoUpdateNewVersionMessage.message,
+      detail: localeMessages.autoUpdateNewVersionInstructions.message,
+      defaultId: RESTART_BUTTON,
+      cancelId: LATER_BUTTON
+    }
+
+    dialog.showMessageBox(options, function(response) {
+      if (response == RESTART_BUTTON) {
+        autoUpdater.quitAndInstall();
+      }
+    });
+  }
+}
+
+function onError(error) {
+  console.log("Got an error while updating: ", error);
```

Comment:
I think here we want to use `error.stack` instead, so we get more information about the error.